snuba/state/__init__.py

```diff
@@ -325,6 +325,12 @@ def record_query(query_metadata: Mapping[str, Any]) -> None:
         logger.exception("Could not record query due to error: %r", ex)
 
 
+def flush_producer() -> None:
+    global kfk
+    if kfk is not None:
+        kfk.flush()
```

Comment:
do we not want to give this a timeout?

Reply:
What timeout do you suggest? In arroyo producer we don't so I think it's fine https://github.com/getsentry/arroyo/blob/4854c67b5ad19c304e3301eff98c45ccaed93a5e/arroyo/backends/kafka/consumer.py#L681